### PR TITLE
Add information about new file opening behaviour

### PR DESCRIPTION
**Pull request by DragaDoncila**
# Description
Add some detail about the new changes to the file opening behaviour to the release notes.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: docs/release/release_0_4_16.md [hunk 1]
```diff
@@ -14,6 +14,35 @@ rendering), and the scientific Python stack (numpy, scipy).
 For more information, examples, and documentation, please visit our
 [website](https://napari.org).
 
+## File Opening Changes in 0.4.16
+
+Prior to `npe2`, file opening with plugins worked through a cascade of function calls trying different readers until one worked, or all failed, in which case an error would be raised. Preferences for readers could be set by reordering hook implementations in the Call Order preference dialog.
+
+This behavior was slow, confusing, and often led to unexpected results. You can see more discussion on this in issue [#4000](https://github.com/napari/napari/issues/4000). `npe2` supports readers declaring a list of accepted filename patterns, and PR [#3799] added a dialog for users to select a plugin to read their file (if more than one was available), and save a preference for that file extension.
```
Comment (Carreau) [inline docs/release/release_0_4_16.md]:
```suggestion
This behavior was slow, confusing, and often led to unexpected results. You can see more discussion on this in issue [#4000](https://github.com/napari/napari/issues/4000). `npe2` supports readers declaring a list of accepted filename patterns, and PR [#3799](https://github.com/napari/napari/pull/3799) added a dialog for users to select a plugin to read their file (if more than one was available), and save a preference for that file extension.
```

**Review by Carreau — APPROVED** [review]
+1, merging soon once CI passes again aften link update.

File: docs/release/release_0_4_16.md [hunk 2]
```diff
@@ -14,6 +14,35 @@ rendering), and the scientific Python stack (numpy, scipy).
...
+
+Before removing plugin call order, we want to ensure that file opening behavior is unified across the GUI and the command line, and that users have access to a smooth workflow for choosing plugins and saving preferences.
+
+After discussion in [#4102](https://github.com/napari/napari/pull/4102) and [#4111](https://github.com/napari/napari/discussions/4111), we decided that as a guiding principle, napari will not choose a reader when multiple are compatible with the given file path. This decision has led to the following changes:
+
+
+- Calling `viewer.open` *without* passing a plugin will result in an error if you have not saved a reader preference for that file pattern *and* multiple plugins can claim the file
+    - You can address this error by associating a preference for the file pattern, or calling `viewer.open(file_path, plugin=...)
```
Comment (brisvag) [inline docs/release/release_0_4_16.md:29]: There's a missing ` here, right?. Maybe @Carreau can add this to #4517 ?

Comment (Carreau) [reply docs/release/release_0_4_16.md:29]: done.

**Pull request merged**